File: backend/src/main/java/com/poeflipfinder/backend/entity/DivinationCardReward.java
package com.poeflipfinder.backend.entity;

/**
 * A divination card's turn-in reward, manually captured and classified per
 * docs/DATA_SOURCES.md. isPredictable = false for gamble/random-reward cards
 * excluded from Feature C (see docs/PRD.md § 7.3) -- kept, not omitted, so
 * the exclusion decision stays visible.
 */
public record DivinationCardReward(
        Currency card,
        int stackSize,
        Currency rewardCurrency,
        int rewardQuantity,
        boolean isPredictable) {
}

File: app/lib/poe_flip_finder/divination_card_reward.ex
defmodule PoeFlipFinder.DivinationCardReward do
  @moduledoc """
  A divination card's turn-in reward, manually captured and classified per
  docs/DATA_SOURCES.md. is_predictable? = false for gamble/random-reward
  cards excluded from Feature C (see docs/PRD.md § 7.3) -- kept, not
  omitted, so the exclusion decision stays visible.
  """

  alias PoeFlipFinder.Currency

  @enforce_keys [:card, :stack_size, :reward_currency, :reward_quantity, :predictable?]
  defstruct [:card, :stack_size, :reward_currency, :reward_quantity, :predictable?]

  @type t :: %__MODULE__{
          card: Currency.t(),
          stack_size: pos_integer(),
          reward_currency: Currency.t(),
          reward_quantity: pos_integer(),
          predictable?: boolean()
        }
end
